File: go/euler-005.go
// 2520 is the smallest number that can be divided by each of
// the numbers from 1 to 10 without any remainder.

// What is the smallest positive number that is
// evenly divisible by all of the numbers from 1 to 20?

// answer: 232792560
// time: .34s / 0.002s

package main

import "fmt"

func test(a int) bool {
	for i := 19; i > 10; i-- {
		if a%i != 0 {
			return false
		}
	}
	return true
}

func main() {
	a := 2520
	for a = 2520; ; a += 20 {
		if test(a) {
			break
		}
	}
	fmt.Println(a)
}

File: js/euler-005.js
// 2520 is the smallest number that can be divided by each of 
// the numbers from 1 to 10 without any remainder.

// What is the smallest positive number that is 
// evenly divisible by all of the numbers from 1 to 20?

// answer: 232792560
// time: 5.318s

// js time: .65s

isHyperDivisible = require('./hyperDivisible');

function smallestHyperDivisible(limit) {
  var number = 2;

  while (!isHyperDivisible(limit, number)) {
    number = number + 2;
  }

  return number;
}


console.log(smallestHyperDivisible(20));